### DISCUSSION TITLE
How to display over formatted help

### DISCUSSION THREAD

**Discussion by PMassicotte** [R-nvim/R.nvim#164] [Ideas]
I am using `lspsaga` to enhance LSP in neovim (https://github.com/nvimdev/lspsaga.nvim).

One interesting feature is the `over` (https://nvimdev.github.io/lspsaga/hover/) function (`shift+k`) that provides function help. When I use this function in other language, the display looks like this:

In Rust:

![image](https://github.com/R-nvim/R.nvim/assets/4519221/8f5f61e2-5019-41d4-bbf3-bf37dd0e3e60)

In Python:

![image](https://github.com/R-nvim/R.nvim/assets/4519221/31f52df8-99dd-4f32-aa65-d73c0bdb793e)

However, while this kinda works with R, the information is displayed as Markdown/HTML:

![image](https://github.com/R-nvim/R.nvim/assets/4519221/01240635-0d6b-4771-9e35-c04973e39f3b)

Anyone know if this could be easlily fixed?

**Comment by jalvesaq** [comment]
It seems that the `languageserver` uses `rmarkdown` to convert the help from HTML to markdown (using the filter `html-to-markdown.lua`):

https://github.com/REditorSupport/languageserver/blob/9b300181f2588462cb5a6f7ecd55fcbeb083d028/R/utils.R#L621

And `rmarkdown` uses `pandoc`:

https://github.com/rstudio/rmarkdown/blob/4b2f3426c1f46a8d207b0d661fc51c2508ecdad7/R/pandoc.R#L40

So, the filter https://github.com/REditorSupport/languageserver/blob/master/inst/lua/html-to-markdown.lua needs improvements.

**Comment by PMassicotte** [comment]
Great, I forgot it was an LSP issue!
